### PR TITLE
Correct handling notifications for iframe

### PR DESCRIPTION
**Pull request by AleshaOleg**

### CODE REVIEW

**Comment by AleshaOleg** [comment]
@sk7 I didn't have time to document behaviour here - https://github.com/ubergrape/grape-web-client/pull/1328/files#diff-e39ae2d33fae1ba175d56e8542e1f0bf0ab39b8d5a7d711ddd3af53216283cc8R295

Notifications for iframe will work only for same domain. Should be written somewhere in iframe documentation here - https://github.com/ubergrape/chatgrape/blob/b0ca670d3630f14d622257b501875642c53ced6f/doc/source/development/embedded_web_client.md#iframe

File: packages/grape-web-client/src/components/browser-notification/BrowserNotification.js [hunk 1]
```diff
@@ -244,19 +246,21 @@ class BrowserNotification extends PureComponent {
...
-          {
-            type: 'grapeClient.updateNotification',
-            payload: {
-              args: {
-                type,
-                properties,
-                callbacks,
-                params,
+          JSON.parse(
+            JSON.stringify({
```
Comment (sk7) [inline packages/grape-web-client/src/components/browser-notification/BrowserNotification.js:250]: why is `JSON.parse(JSON.stringify(` necessary? looks like a hack :D

File: packages/grape-web-client/demo/iframe.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    #grape-client {
+      position: absolute;
+      width: 100%;
+      height: 100%;
+    }
+  </style>
+</head>
+<body>
+  <iframe src="https://db2fe37f28fa.ngrok.io/chat/" id="grape-client" />
```
Comment (sk7) [inline packages/grape-web-client/demo/iframe.html:22]: this ngrok URL is dead already, maybe put staging or a alpha.chatgrape.com there?

File: packages/grape-web-client/src/components/browser-notification/BrowserNotification.js [hunk 3]
```diff
@@ -286,21 +290,26 @@ class BrowserNotification extends PureComponent {
       conf,
     })
 
-    if (window.top !== window.self) {
+    if (
+      window.top !== window.self &&
+      getDomain(document.referrer) === getDomain(window.self.location.href)
```
Comment (sk7) [inline packages/grape-web-client/src/components/browser-notification/BrowserNotification.js:295]: but the iframing domain will be different from the iframe domain in the case of Untis?

Comment (AleshaOleg) [reply packages/grape-web-client/src/components/browser-notification/BrowserNotification.js:295]: Yes and no :D The subdomain will be different, that's why switching to staging will not work in previous comment

**Pull request merged**